tensorrt_llm/_torch/distributed/ops.py

```
@@ -222,6 +222,168 @@ def convert_output(x, x_info):
+    assert len(dims) == len(inputs)
+    assert all(isinstance(dim, int) for dim in dims)
+    assert len(new_dims) == len(inputs)
+    assert all(new_dim is None or isinstance(new_dim, int)
+               for new_dim in new_dims)
+
+    op_inputs = []
+
+    for inp, dim, new_dim in zip(inputs, dims, new_dims):
+        size_per_rank, rem = divmod(inp.shape[dim], n_ranks)
+        assert rem == 0, \
+            f"input.shape[{dim}] must be divisible by n_ranks ({n_ranks}), but got shape {inp.shape} for rank {mapping.tp_rank}"
```

Ruff (0.12.2)

Undefined name `mapping`

(F821)